src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorBuilder.cs

```diff
@@ -42,23 +42,21 @@ public static TagHelperDescriptorBuilder Create(string kind, string name, string
             return new DefaultTagHelperDescriptorBuilder(kind, name, assemblyName);
         }
 
-        public abstract IDictionary<string, string> Metadata { get; }
-
-        public abstract TagHelperDescriptorBuilder BindAttribute(Action<BoundAttributeDescriptorBuilder> configure);
+        public abstract string DisplayName { get; set; }
 
-        public abstract TagHelperDescriptorBuilder TagMatchingRule(Action<TagMatchingRuleDescriptorBuilder> configure);
+        public abstract string TagOutputHint { get; set; }
 
-        public abstract TagHelperDescriptorBuilder AllowChildTag(string allowedChild);
+        public abstract string Documentation { get; set; }
 
-        public abstract TagHelperDescriptorBuilder TagOutputHint(string hint);
+        public abstract ICollection<string> AllowedChildTags { get; }
```

Comment:
Thinking we might want to make this a custom collection to protect ourselves. @rynowak thoughts?